devimint/src/external.rs

```diff
@@ -976,65 +980,26 @@ pub async fn open_channels_between_gateways(
         gateways.iter().circular_tuple_windows::<(_, _)>().collect()
     };
 
-    let open_channel_tasks = gateway_pairs.iter()
-        .map(|((gw_a, gw_a_name), (gw_b, gw_b_name))| {
-            let gw_a = (*gw_a).clone();
-            let gw_a_name = (*gw_a_name).to_string();
-            let gw_b = (*gw_b).clone();
-            let gw_b_name = (*gw_b_name).to_string();
-
-            let sats_per_side = 5_000_000;
-            debug!(target: LOG_DEVIMINT, from=%gw_a_name, to=%gw_b_name, "Opening channel with {sats_per_side} sats on each side...");
-            tokio::task::spawn(async move {
-                // Sometimes channel openings just after funding the lightning nodes don't work right away.
-                let res = poll_with_timeout(&format!("Open channel from {gw_a_name} to {gw_b_name}"), Duration::from_secs(30), || async {
-                    gw_a.open_channel(&gw_b, sats_per_side * 2, Some(sats_per_side)).await.map_err(ControlFlow::Continue)
-                })
-                .await;
-
-                if res.is_err() {
-                    error!(target: LOG_DEVIMINT, from=%gw_a_name, to=%gw_b_name, ?res, "Failed to open channel");
-                    gw_a.dump_logs()?;
-                    gw_b.dump_logs()?;
-                }
-
-                res
-            })
+    let sats_per_side = 5_000_000;
+    for ((gw_a, gw_a_name), (gw_b, gw_b_name)) in &gateway_pairs {
+        // Sometimes channel openings just after funding the lightning nodes don't work
+        // right away.
+        let txid = poll(&format!("Open channel from {gw_a_name} to {gw_b_name}"), || async {
+            info!(target: LOG_DEVIMINT, from=%gw_a_name, to=%gw_b_name, "Opening channel with {sats_per_side} sats on each side...");
+            gw_a.open_channel(gw_b, sats_per_side * 2, Some(sats_per_side)).await.map_err(ControlFlow::Continue)
         })
-        .collect::<Vec<_>>();
-    let open_channel_task_results: Vec<Result<Result<_, _>, _>> =
-        futures::future::join_all(open_channel_tasks).await;
-
-    let mut channel_funding_txids = Vec::new();
-    for open_channel_task_result in open_channel_task_results {
-        match open_channel_task_result {
-            Ok(Ok(txid)) => {
-                channel_funding_txids.push(txid);
-            }
-            Ok(Err(e)) => {
-                return Err(anyhow::anyhow!(e));
-            }
-            Err(e) => {
-                return Err(anyhow::anyhow!(e));
-            }
-        }
-    }
+        .await?;
 
-    // Wait for all channel funding transaction to be known by bitcoind.
-    let mut is_missing_any_txids = false;
-    for txid_or in &channel_funding_txids {
-        if let Some(txid) = txid_or {
-            bitcoind.poll_get_transaction(*txid).await?;
-        } else {
-            is_missing_any_txids = true;
+        if let Some(txid) = txid {
+            bitcoind.poll_get_transaction(txid).await?;
```

Comment:
Is waiting for the tx enough or should we wait for a specific number of confirmations actually so the LN node recognizes the change as spendable again?

Reply:
Waiting is enough, we just want to make sure it is in the mempool. Blocks are mined a bit below.